src/lazyload-image.ts

```diff
@@ -110,15 +120,15 @@ function setLoadedStyle(element: HTMLImageElement | HTMLDivElement) {
     return element;
 }
 
-export function lazyLoadImage(element: HTMLImageElement | HTMLDivElement, imagePath: string, defaultImagePath: string, errorImgPath: string, offset: number, useSrcset: boolean = false) {
+export function lazyLoadImage(element: HTMLImageElement | HTMLDivElement, imagePath: string, defaultImagePath: string, errorImgPath: string, offset: number, useSrcset: boolean = false, scrollContainer?: HTMLElement) {
     setImageAndSourcesToDefault(element, defaultImagePath, useSrcset);
     if (element.className && element.className.includes('ng-lazyloaded')) {
         element.className = element.className.replace('ng-lazyloaded', '');
     }
 
     return (scrollObservable: Observable<Event>) => {
         return scrollObservable
-            .filter(() => isVisible(element, offset))
+            .filter(() => isVisible(element, offset, window, scrollContainer))
```

Comment:
I chose to pass the window object here, which might not be the pretties thing to do. But I didn't want to break the argument order.
Would you rather have me flip the arguments, so that `_window = window` and I change all of the tests to comply with the change?

Reply:
This is fine. It feels however a bit weird to have an optional argument (`threshold`) before a mandatory (`_window`) but I think we should keep `threshold = 0` so it defaults to 0 if `threshold` is `undefined`, so keep it as it is.